lib/hiera/backend/eyaml/encryptors/gpg.rb

```diff
@@ -17,7 +17,7 @@ class Gpg < Encryptor
           self.options = {
             :gnupghome => { :desc => "Location of your GNUPGHOME directory",
                             :type => :string,
-                            :default => "#{ENV[["HOME", "HOMEPATH"].detect { |h| ENV[h] != nil }]}/.gnupg" },
+                            :default => ENV["HOME"] || ENV["HOMEPATH"] || ".gnupg" },
```

Comment:
Agreed, that would be a better approach.
Check for $HOME/.gnupg $HOMEPATH/.gnupg, $GNUPGHOME, throw error if otherwise